app/models/provider_relationship_permissions.rb

```diff
@@ -2,8 +2,24 @@ class ProviderRelationshipPermissions < ApplicationRecord
   belongs_to :ratifying_provider, class_name: 'Provider'
   belongs_to :training_provider, class_name: 'Provider'
 
-  VALID_PERMISSIONS = %i[view_safeguarding_information make_decisions].freeze
+  # For ease of use in views, this struct acts like a ProviderPermissions model
+  PermissionsSet = Struct.new(:make_decisions?, :view_safeguarding_information?) do
```

Comment:
This is fine for display purposes, but considering we'll be passing form objects to components to handle updates, perhaps we should make our default org permissions interface return form objects rather than structs. As in, `#ratifying_provider_permissions` would return a `RatifyingProviderPermissions` ActiveModel-backed object.

On a more general note, it seems to me we're trying to simplify those complex unwieldy new column names (`training_provider_can_make_decisions`) and start using simpler permissions names like `make_decisions` in the code again. Wasn't the original refactor intention to be as specific as possible with the column/method names? Another direction would be to pass the actual permissions model to components, but use different components for showing ratifying and training provider org permissions. I think I would prefer this.

Reply:
My understanding is that the form will deal with and validate a pair of permissions at a time, so there's no need to carve out separate form objects, eg https://manage-applications-beta.herokuapp.com/organisations/edit. That form will deal directly with `ratifying_provider_can` et al.

The struct is just so we can use the same template/logic to render permissions on either side of the relationship, as this is the same on both sides, eg: https://manage-applications-beta.herokuapp.com/organisations/show3.

Reply:
The form object suggestion was just an exploration to see if we can return something a bit more useful than the structs, if `#ratifying_provider_permissions` and `#training_provider_permissions` become the main way of accessing these subsets for a given relationship.

My main reservation is these structs could become an interface for checking permissions, along the lines of ```relationship.ratifying_provider_permissions.make_decisions?```, which doesn't actually look that bad written like this but could become bug-prone if the struct itself is passed to another part of the code, where `make_decisions?` will have lost some of its context. As long as anything decision-related goes via `ProviderAuthorisation`, this is not an issue, of course, but I'd prefer the structs to be generated in the component code that uses them, rather than the model, if their main purpose is to allow components that work seamlessly for both training and ratifying provider permissions.

As already discussed on a call, I'm happy for this to be merged as is and perhaps refactor the way components are made to work for both training and ratifying provider permissions.

Reply:
@mnacos Thanks for writing this up! We will revisit after @AgaDufrat's next PR is in.